nuxeo-distribution/nuxeo-jsf-ui-hotreload-tests/src/test/java/org/nuxeo/ftest/jsf/hotreload/studio/NuxeoITCase.java

```diff
@@ -0,0 +1,193 @@
+/*
+ * (C) Copyright 2017 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Anahide Tchertchian
+ */
+package org.nuxeo.ftest.jsf.hotreload.studio;
+
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.nuxeo.functionaltests.AjaxRequestManager;
+import org.nuxeo.functionaltests.Locator;
+import org.nuxeo.functionaltests.RestHelper;
+import org.nuxeo.functionaltests.forms.LayoutElement;
+import org.nuxeo.functionaltests.pages.AbstractPage;
+import org.nuxeo.functionaltests.pages.DocumentBasePage;
+import org.nuxeo.functionaltests.pages.DocumentBasePage.UserNotConnectedException;
+import org.nuxeo.functionaltests.pages.forms.FileCreationFormPage;
+import org.nuxeo.functionaltests.pages.search.SearchPage;
+import org.openqa.selenium.By;
+import org.openqa.selenium.NoSuchElementException;
+import org.openqa.selenium.WebElement;
+
+import static org.nuxeo.functionaltests.Constants.WORKSPACES_PATH;
+
+import static org.junit.Assert.assertEquals;
+
+/**
+ * Test case for testing Studio generated jars on Nuxeo, holds all tests that should be run on most of target platforms
+ * (CAP and DM for now)
+ *
+ * @since 2.9
+ */
+public abstract class NuxeoITCase extends NuxeoITCaseHelper {
+
+    private static final String USERNAME = "gudule";
+
+    private static final String PASSWD = "gudule1";
+
+    @Before()
+    public void before() {
+        RestHelper.createUser(USERNAME, PASSWD, "firstname1", "lastname1", "company1", "email1", "members");
+        RestHelper.addPermission(WORKSPACES_PATH, USERNAME, "Write");
+    }
+
+    @After
+    public void after() {
+        logout();
+        RestHelper.deleteDocument(WORKSPACES_PATH + TEST_WS_TITLE);
+        RestHelper.removePermissions(WORKSPACES_PATH, USERNAME);
+        RestHelper.cleanup();
+    }
+
+    protected String getTestCaseId() {
+        return this.getClass().getSimpleName();
+    }
+
+    protected StudioDocumentPage getTestStudioDocument() {
+        StudioDocumentPage d = asPage(StudioDocumentPage.class);
+        d.setTestId(getTestCaseId());
+        return d;
+    }
+
+    protected void doLog() throws UserNotConnectedException {
+        login(USERNAME, PASSWD);
```

Comment:
indeed, note these are copy/pasted from Studio code: not sure about history here, but makes sense to change